modules/programs/hexchat.nix

```diff
@@ -0,0 +1,324 @@
+          type = nullOr (enum [
+            "nickServMsg"
+            "nickServ"
+            "challengeAuth"
+            "sasl"
+            "serverPassword"
+            "saslExternal"
+            "customCommands"
+          ]);
+
+          description = ''
+            null => Default
+            "nickServMsg" (1) => NickServ (/MSG NickServ + password)
+            "nickServ" (2) => NickServ (/NICKSERV + password)
+            "challengeAuth" (4) => Challenge Auth (username + password)
+            "sasl" (6) => SASL (username + password)
+            "serverPassword" (7) => Server password (/PASS password)
+            "saslExternal" (10) => SASL EXTERNAL (cert)
+            "customCommands" (9) => Custom (Use "commands" field for Auth, like: 'commands = [ "/msg NickServ IDENTIFY my_password" ];' )
+          '';
+
+          default = null;
+        };
+
+        nickname = mkOption {
+          type = nullOr str;
+          default = null;
+          description = "Primary nickname.";
+        };
+
+        nickname2 = mkOption {
+          type = nullOr str;
+          default = null;
+          description = "Secondary nickname.";
+        };
+
+        options = mkOption {
+          type = nullOr channelOptions;
+          default = null;
+          description = "Channel options.";
+          example = literalExample
+            "{ autoconnect = true; useGlobalUserInformation = true; }";
+        };
+
+        password = mkOption {
+          type = nullOr str;
+          default = null;
+          description = "Password.";
+        };
+
+        realName = mkOption {
+          type = nullOr str;
+          default = null;
+          description = "Real name.";
+        };
+
+        servers = mkOption {
+          type = listOf str;
+          description = "IRC Server Address List.";
+          example =
+            literalExample ''[ "chat.freenode.net" "irc.freenode.net" ]'';
+          default = [ ];
+        };
+
+        userName = mkOption {
+          type = nullOr str;
+          default = null;
+          description = "User name.";
+        };
+      };
+    };
+
+  # helpers
+  transformField = k: v: if (v != null) then "${k}=${v}" else null;
+
+  listChar = c: l:
+    if (l != [ ]) then
+      (concatMapStringsSep "\n" (transformField c) l)
+    else
+      null;
+
+  computeFieldsValue = (channel:
+    (toString (if channel.options == null then
+      0
+    else
+      (with channel.options;
+        (if autoconnect then 8 else 0)
+        + (if !connectToSelectedServerOnly then 1 else 0)
+        + (if !bypassProxy then 16 else 0) + (if forceSSL then 4 else 0)
+        + (if acceptInvalidSSLCertificates then 32 else 0)
+        + (if useGlobalUserInformation then 2 else 0)))));
+
+  loginMethod = let
+    loginMethodMap = {
+      nickServMsg = 1;
+      nickServ = 2;
+      challengeAuth = 4;
+      sasl = 6;
+      serverPassword = 7;
+      saslExternal = 10;
+      customCommands = 9;
+    };
```

Comment:
Why not move this at the top and set `modChannelOption.loginMethod.type = enum (attrValues loginMethodMap)`?